### ISSUE TITLE
AM08 and Redshift natural joins

### ISSUE THREAD

**Issue by natsirtguy** [sqlfluff/sqlfluff#6507]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

SQLFluff had a false positive AM08 (Implicit cross join detected) for a natural join in Redshift (see [here](https://docs.aws.amazon.com/redshift/latest/dg/r_Join_examples.html) for documentation).

Example query:
```
select foo.x
from foo
natural join bar
```

Output:
```
L:   3 | P:   1 | AM08 | Implicit cross join detected. [ambiguous.join_condition]
```

### Expected Behaviour

No AM08 detection **or** perhaps another rule that recommends against natural joins entirely (since the columns used in the join condition are implicit).

### Observed Behaviour

Example query:
```
select foo.x
from foo
natural join bar
```

Output:
```                                                                          
L:   3 | P:   1 | AM08 | Implicit cross join detected.
                       | [ambiguous.join_condition]
```

### How to reproduce

Create a file with
```
select foo.x
from foo
natural join bar
```
in it and run `sqlfluff lint <file> --dialect redshift`

### Dialect

Redshift

### Version

sqlfluff, version 3.3.0
Python 3.12.6

### Configuration

.sqlfluff
```
[sqlfluff]
dialect = redshift
max_line_length = 100
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by rogalski** [comment]
@natsirtguy Thanks for the report.

We already ignore positional joins from DuckDB or JOIN ... USING. This falls into same bucket, I'll prep a fix.
